src/zjs_util.h

```diff
@@ -56,6 +56,35 @@ bool zjs_obj_get_double(jerry_value_t obj, const char *name, double *num);
 bool zjs_obj_get_uint32(jerry_value_t obj, const char *name, uint32_t *num);
 bool zjs_obj_get_int32(jerry_value_t obj, const char *name, int32_t *num);
 
+/**
+ * Copy a JerryScript string into a supplied char * buffer.
+ *
+ * @param jstr    A JerryScript string value.
+ * @param buffer  A char * buffer with at least maxlen + 1 bytes.
+ * @param maxlen  Pointer to a maximum size for the copied string. The buffer
+ *                  allocated may be larger, but the string will be no larger
+ *                  than this (not including null terminator which is guaranteed
+ *                  to be there). If NULL or smaller than the actual string
+ *                  size, only the null terminator will be copied to the string.
+ *                  If not NULL, the actual copied length of the string will be
+ *                  written to *maxlen.
+ */
+void zjs_copy_jstring(jerry_value_t jstr, char *buffer, jerry_size_t *maxlen);
+
+/**
+ * Allocate a char * buffer on the heap and copy the JerryScript string to it.
+ *
+ * @param jstr    A JerryScript string value.
+ * @param maxlen  Pointer to a maximum size for the returned string. The buffer
+ *                  allocated may be larger, but the string will be no larger
+ *                  than this (not including null terminator which is guaranteed
+ *                  to be there). If NULL or pointing to 0, there is no limit
```

Comment:
I personally think you should specifically call out that if jstr is longer than maxlen, it will be truncated.  I think it would help make it clearer.